src/sysusers/sysusers.c

```diff
@@ -1209,12 +1209,23 @@ static int process_item(Item *i) {
 
         switch (i->type) {
 
-        case ADD_USER:
-                r = add_group(i);
-                if (r < 0)
-                        return r;
+        case ADD_USER: {
+                Item *j;
+
+                j = ordered_hashmap_get(groups, i->name);
+                if (j && j->todo_group) {
+                        /* The group with the same name is already in queue.
+                         * So, let's use the information about the group. */
+                        i->gid_set = j->gid_set;
```

Comment:
This is still failing because "add_user()" will do a uid_is_ok()" check which will prevent different uid/groupid. This means that user "mail" gets 12:12 because the code will reuse the group id. I think this is expected behaviour and consistent with what systemd-sysusers was doing before we added the extended "uid:gid" syntax. So I think this PR just needs a test update to "mail:x:12:12..:".

To get 8:12 it would be possible to set `i->id_set_strict = true` here. I tested this locally and with that all existing tests pass. However this also will cause unwanted side effects (i.e. if the uid is already taken it will create wrong entries now). So we would need a way to specify that we only want the gid to be fixed.

Reply:
Please let me know if I can help in any way, happy to take your PR and tweak the test (but given how trivial that is it is probably not worth it).

Reply:
> To get 8:12 it would be possible to set i->id_set_strict = true here. 

Good catch! Thanks.

Reply:
> However this also will cause unwanted side effects (i.e. if the uid is already taken it will create wrong entries now).

Hmm? Could you give me an example for this?

Reply:
You are right, I think this is actually ok - I was misrembering what id_set_strict checks - but it only checks for group id clashes so that is fine and we can add it. I also think its the right thing to do and in line what the users expect. But AFAICT it's a behaviour change from the old code (even from the code from before uid:gid) because that code would detect the uid/gid name match and also match the uid/gid. However I think its ok to argue this is a bugfix. So from my POV adding i->id_set_strict = true is fine. We might release note it (that something changed in this particular corner case).